### PR TITLE
DEVPROD-10201: Improve mapping between GraphQL User type and APIDBUser struct

### PR DESCRIPTION
**Pull request by minnakt**
DEVPROD-10201

### Description
I noticed that the way in which we process a `User` in GraphQL doesn't follow the patterns we have elsewhere. This PR updates the logic around `APIDBUser` which is the underlying model for `User` in GraphQL.

### Testing
- Existing tests should pass
- Tested via `/graphql` sandbox

### CODE REVIEW

File: rest/model/generated.go
Comment (minnakt) [inline rest/model/generated.go:1]: this is legacy generated code but according to the comments manual edits are allowed? The legacy codegen tool was already removed from Evergreen a while ago

File: rest/data/user.go [hunk 1]
```diff
@@ -171,6 +173,10 @@ func AddOrUpdateServiceUser(toUpdate restModel.APIDBUser) error {
 		return errors.New("cannot update an existing non-service user")
 	}
 	toUpdate.OnlyApi = true
-	dbUser := restModel.APIDBUserToService(toUpdate)
+
+	dbUser, err := toUpdate.ToService()
+	if err != nil {
```
Comment (hadjri) [inline rest/data/user.go:178]: nit: nil check dbUser

File: graphql/user_resolver.go [hunk 2]
```diff
@@ -6,45 +6,10 @@ import (
 
 	"github.com/evergreen-ci/evergreen/model/event"
 	"github.com/evergreen-ci/evergreen/model/patch"
-	"github.com/evergreen-ci/evergreen/model/user"
 	restModel "github.com/evergreen-ci/evergreen/rest/model"
 	"github.com/evergreen-ci/utility"
 )
 
-// ParsleyFilters is the resolver for the parsleyFilters field.
```
Comment (hadjri) [inline graphql/user_resolver.go:14]: Is this removal intended? Was the removal generated?

Comment (minnakt) [reply graphql/user_resolver.go:14]: now that we use the complete `APIDBUser` as the underlying model, these get automatically resolved so we don't have to write any resolver code for them!

**Review by hadjri — APPROVED** [review]

**Review by sophstad — APPROVED** [review]
